### ISSUE TITLE
CI: add String() function for CmdRes type and consume it in the wrappers for expectations of CmdRes output

### ISSUE THREAD

**Issue by ianvernon** [cilium/cilium#3469]
Right now, some of the messages which are output when a test fails are cryptic and sometimes don't give enough information to see exactly what happened when state that is being analyzed is not what is expected. Being able to quickly see all metadata associated with a given `CmdRes` as a structured string would be extremely useful for debugging purposes in such situations (i.e. the command passed into it, parameters, stdout, stderr, and its exit code). And, as part of the wrapper functions for checking the success / failure of commands (e.g., `ExpectSuccess`), we can use this `String()` function for seeing what command was ran, what its return code was, etc., with no additional configuration needed on the test-writer's part for debugging.

**Comment by eloycoto** [comment]
At the moment we have the following: 
```
// GetDebugMessage returns executed command and its output
func (res *CmdRes) GetDebugMessage() string {
	return fmt.Sprintf("cmd: %s\noutput: %s", res.GetCmd(), res.CombineOutput())
}
```

That has all the information that you want ;-)

**Comment by joestringer** [comment]
Maybe addressed by #4087?

**Comment by eloycoto** [comment]
Closed by #4087
